sktree/tree/_multiview.py

```diff
@@ -360,6 +378,51 @@ def _build_tree(
         else:
             SPLITTERS = DENSE_SPLITTERS
 
+        if self.apply_max_features_per_feature_set:
+            # XXX: experimental
+            # we can replace max_features_ here based on whether or not uniform logic over
+            # feature sets
+            max_features_per_set = []
+            n_features_in_prev = 0
+            for idx in range(self.n_feature_sets_):
+                n_features_in_ = self.feature_set_ends_[idx] - n_features_in_prev
+                n_features_in_prev = n_features_in_
+                import numbers
+
+                if isinstance(self.max_features, str):
+                    if self.max_features == "sqrt":
+                        max_features = max(1, int(np.sqrt(n_features_in_)))
+                    elif self.max_features == "log2":
+                        max_features = max(1, int(np.log2(n_features_in_)))
+                elif self.max_features is None:
+                    max_features = n_features_in_
+                elif isinstance(self.max_features, numbers.Integral):
+                    max_features = self.max_features
+                else:  # float
+                    if self.max_features > 0.0:
+                        max_features = max(1, int(self.max_features * n_features_in_))
+                    else:
+                        max_features = 0
+
+                if max_features > n_features_in_:
+                    raise ValueError(
+                        f"max_features must be less than or equal to "
+                        f"the number of features in feature set {idx}: {n_features_in_}, but "
+                        f"max_features = {max_features} when applying sampling"
+                        f"per feature set."
+                    )
+
+                max_features_per_set.append(max_features)
+            self.max_features_ = np.sum(max_features_per_set)
+            if self.max_features_ > n_features:
+                raise ValueError(
+                    "max_features is greater than the number of features. "
```

Comment:
I think the error trace should write out the actual values of `max_features` and `n_features` to be more useful

Reply:
Added.